corehq/apps/change_feed/consumer/feed.py

```diff
@@ -80,9 +80,6 @@ def iter_changes(self, since, forever):
             offsets = [
                 copy(self._processed_topic_offsets)
             ]
-            topics_missing = set(self._topics) - checkpoint_topics
-            for topic in topics_missing:
-                offsets.append(topic)  # consume all available partitions
```

Comment:
documented in the commit message https://github.com/dimagi/commcare-hq/pull/16020/commits/2107e8ee5a7319712d77edc2dd6cd45ded751d0d